File: Server/src/GlobalsManager.ts
import { HashUtil } from "@spt/utils/HashUtil"
import { AbstractModManager } from "./AbstractModManager"
import { Constants } from "./Constants"

export class GlobalsManager extends AbstractModManager
{
    protected configName: string = "GlobalsConfig"

    protected afterPostDB(): void
    {
        this.logger.debug(this.jsonUtil.serialize(this.databaseTables.globals.ItemPresets))

        if (Constants.PrintNewHashes)
        {
            const hashUtil = this.container.resolve<HashUtil>("HashUtil")
    
            for (let i = 0; i < 100; i++)
            {
                console.log(hashUtil.generate())
            }
        }

        if (this.config.removeStatusEffectRemovePrices == true)
        {
            this.databaseTables.globals.config.Health.Effects.BreakPart.RemovePrice = 0
            this.databaseTables.globals.config.Health.Effects.Fracture.RemovePrice = 0
            this.databaseTables.globals.config.Health.Effects.LightBleeding.RemovePrice = 0
            this.databaseTables.globals.config.Health.Effects.HeavyBleeding.RemovePrice = 0
            this.databaseTables.globals.config.Health.Effects.Intoxication.RemovePrice = 0
        }

        if (this.config.removePostRaidHeal == true)
        {
            for (let traderKey in this.databaseTables.traders)
            {
                const trader = this.databaseTables.traders[traderKey]

                trader.base.medic = false
            }
        }
        
        this.databaseTables.globals.config.RagFair.enabled = !(this.config.disableFlea == true)

        if (this.config.disableFleaBlacklist == true)
        {
            for (let itemKey in this.databaseTables.templates.items)
            {
                const item = this.databaseTables.templates.items[itemKey]

                item._props.CanSellOnRagfair = true
                item._props.CanRequireOnRagfair = true
            }
            
        }

        if (this.config.removeTraderMoneyRequirements == true)
        {
            for (let traderKey in this.databaseTables.traders)
            {
                const trader = this.databaseTables.traders[traderKey]

                trader.base.loyaltyLevels.forEach(level => 
                {
                    level.minSalesSum = 0
                })
            }
        }

        if (this.config.delayScavRun == true)
        {
            this.databaseTables.globals.config.SavagePlayCooldown = 2147483646
        }

        this.logger.info(`${Constants.ModTitle}: Globals changes applied!`)
    }
}

File: Server/VAI-RLTP/Managers/GlobalsManager.cs
using System;
using SPTarkov.DI.Annotations;

namespace VAI.RLTP.Managers;

[Injectable(InjectionType.Singleton)]
public sealed class GlobalsManager : AbstractModManager
{
    protected override string ConfigName => "GlobalsConfig";

    protected override void AfterPostDb()
    {
        if (Constants.PrintNewHashes)
        {
            TryPrintHashes();
        }

        if (GetConfigBool("removeStatusEffectRemovePrices"))
        {
            var effects = DatabaseTables.Globals?.Configuration?.Health?.Effects;
            if (effects != null)
            {
                effects.BreakPart.RemovePrice = 0;
                effects.Fracture.RemovePrice = 0;
                effects.LightBleeding.RemovePrice = 0;
                effects.HeavyBleeding.RemovePrice = 0;
                effects.Intoxication.RemovePrice = 0;
            }
        }

        if (GetConfigBool("removePostRaidHeal"))
        {
            foreach (var trader in DatabaseTables.Traders.Values)
            {
                trader.Base.Medic = false;
            }
        }

        if (GetConfigBool("disableFlea"))
        {
            var ragfair = DatabaseTables.Globals?.Configuration?.RagFair;
            if (ragfair != null)
            {
                ragfair.Enabled = true;
                ragfair.MinUserLevel = 70;
            }
        }

        if (GetConfigBool("disableFleaBlacklist"))
        {
            foreach (var item in DatabaseTables.Templates.Items.Values)
            {
                if (item.Properties == null)
                {
                    continue;
                }

                item.Properties.CanSellOnRagfair = true;
                item.Properties.CanRequireOnRagfair = true;
            }
        }

        if (GetConfigBool("removeTraderMoneyRequirements"))
        {
            foreach (var trader in DatabaseTables.Traders.Values)
            {
                if (trader.Base?.LoyaltyLevels == null)
                {
                    continue;
                }

                foreach (var level in trader.Base.LoyaltyLevels)
                {
                    level.MinSalesSum = 0;
                }
            }
        }

        if (GetConfigBool("delayScavRun"))
        {
            if (DatabaseTables.Globals?.Configuration != null)
            {
                DatabaseTables.Globals.Configuration.SavagePlayCooldown = 2147483646;
            }
        }

        Constants.GetLogger().Info($"{Constants.ModTitle}: Globals changes applied!");
    }

    private static void TryPrintHashes()
    {
        try
        {
            for (var i = 0; i < 100; i++)
            {
                var value = Helper.GenerateSha256Id(Guid.NewGuid().ToString("N"));
                Console.WriteLine(value);
            }
        }
        catch (Exception ex)
        {
            Constants.GetLogger().Warning($"{Constants.ModTitle}: Failed to print hashes. {ex.Message}");
        }
    }
}
